Source/CBLMultipartReader.m

```diff
@@ -137,17 +138,20 @@ - (BOOL) parseHeaders: (NSString*)headersStr {
 
 
 - (NSRange) searchFor: (NSData*)pattern from: (NSUInteger)start {
-    return [_buffer rangeOfData: pattern
+    if (!_buffer.hasBytesAvailable)
+        return NSMakeRange(NSNotFound, 0);
+    NSData *data = [NSData dataWithBytesNoCopy:_buffer.mutableBytes length:_buffer.bytesAvailable freeWhenDone:NO];
```

Comment:
It'd be better to implement -rangeOfData:... directly on CBLMultipartBuffer. Then it can search without creating a new NSData object since it knows the range to search in.
(This does probably make a difference, since this method is called on every read to find the multipart boundary string.)